### PR TITLE
Closes #51 : Added index.d.ts + documentation for javascript and typescript

### LINKED ISSUE #51: Can you consider writing `type.d.ts`
I used it in `typescript`, can the author write `index.d.ts` for this library, thank you!

### PR DESCRIPTION
**Pull request by iArmann**
### Closes #51 : Added **`index.d.ts`** + **documentation** for _Javascript_ and _Typescript_

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * ## Modes :
+ * Modes are just ready-to-use faces, here's their list:
+ * - #### `b` : borg
+ * - #### `d` : dead      
+ * - #### `g` : greedy
+ * - #### `p` : paranoia
+ * - #### `s` : stoned
+ * - #### `t` : tired
+ * - #### `w` : youthful
+ * - #### `y` : wired
```
Comment (lachieh) [inline index.d.ts:85]:
```suggestion
 * - #### `w` : wired
 * - #### `y` : youthful
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
```
Comment (lachieh) [inline index.d.ts:136]:
```suggestion
 * - #### `w` : wired
 * - #### `y` : youthful
```

File: index.js [hunk 3]
```diff
@@ -2,14 +2,133 @@ var baloon = require("./lib/balloon");
...
+ * ## Modes :
+ * Modes are just ready-to-use faces, here's their list:
+ * - #### `b` : borg
+ * - #### `d` : dead      
+ * - #### `g` : greedy
+ * - #### `p` : paranoia
+ * - #### `s` : stoned
+ * - #### `t` : tired
+ * - #### `w` : youthful
+ * - #### `y` : wired
```
Comment (lachieh) [inline index.js:28]:
```suggestion
 * - #### `w` : wired
 * - #### `y` : youthful
```

File: index.js [hunk 4]
```diff
@@ -2,14 +2,133 @@ var baloon = require("./lib/balloon");
...
```
Comment (lachieh) [inline index.js:83]:
```suggestion
 * - #### `w` : wired
 * - #### `y` : youthful
```

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    T?: string, // tongue
+    // modes
+    b?: boolean, // borg 
+    d?: boolean, // dead
+    g?: boolean, // greedy
+    p?: boolean, // paranoia
+    s?: boolean, // stoned
+    t?: boolean, // tired
+    w?: boolean, // youthful
+    y?: boolean, // wired
```
Comment (lachieh) [inline index.d.ts:38]:
```suggestion
    w?: boolean, // wired
    y?: boolean, // youthful
```

**Review by lachieh — CHANGES_REQUESTED** [review]
This all looks good to be, aside from a few the y and w being switched in some of the examples. I'm not the owner but I have confirmed that this file works as expected. I appreciate the updated readme with examples too.

@piuccio if you're good with the README.md changes, this is good from a TS perspective.

**Review by piuccio — APPROVED** [review]

**Pull request merged**